### DISCUSSION TITLE
Open WebUI LXC

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#3994] [Announcements]
**If you selected to add "Ollama" during the Open WebUI installation**

After setting up your user, all that's needed is to select a model. Click on the drop-down menu, then type `llama3.2:1b` in the "search for model" field (or choose from others available [here](https://ollama.com/library?sort=popular)).

To check if Ollama is running, enter your Open WebUI IP followed by `:11434` in your browser.

**Comment by Deekerman** [comment]
I had to edit the .env for it to work

old
```
OLLAMA_BASE_URL=http://0.0.0.0:11434
```
new
```
OLLAMA_BASE_URL=http://<ip>:11434
```

then it worked

**Reply by tteck** [reply]
`OLLAMA_BASE_URL=http://0.0.0.0:11434` works here 🤷🏻‍♂️

**Reply by tteck** [reply]
@Deekerman is Ollama installed on the Open WebUI LXC or in its own LXC?

**Reply by Deekerman** [reply]
@tteck it was installed through the open webUI script itself. It told me it was unable to connect. then once i changed it, it worked. could be something on my side. but it told me it was unable to connect to 0.0.0.0

is it ok to connect using the ip?

**Reply by Deekerman** [reply]
@tteck Does this have gpu passthrough set up?

**Reply by havardthom** [reply]
It does not have gpu passthrough set up

**Reply by Deekerman** [reply]
thanks. I added my gpu and stuff but wasnt sure how to enable it in the programs or if it would do it on its own. Ill look more. 
Thanks

**Reply by havardthom** [reply]
From what I've gathered:
For Nvidia you need drivers: https://github.com/ollama/ollama/blob/main/docs/linux.md#install-cuda-drivers-optional
For AMD you need additional package and drivers: https://github.com/ollama/ollama/blob/main/docs/linux.md#amd-gpu-install https://github.com/ollama/ollama/blob/main/docs/linux.md#install-amd-rocm-drivers-optional
For Intel you need to build Ollama source code from scratch with intel-basekit and have Intel drivers (this script covers it: https://github.com/tteck/Proxmox/blob/main/install/ollama-install.sh)

**Reply by Deekerman** [reply]
@havardthom Thank you. I have the drivers installed for the gpu, and nvidia-smi pulls up the gpu information. It just doesnt seem to be using it. So i wasnt sure if there was somewhere within open webui or ollama i needed to activate the use. or something specific i need from either of them for it to work. 

For reference, im using a Quadro P2000 
Driver Version: 550.67    CUDA Version: 12.4

**Reply by havardthom** [reply]
If run these in LXC console you should see some logs on ollama startup like which LLM runners it supports, GPU discovery and which device is used for inference.
```
systemctl restart ollama.service
journalctl -u ollama.service -n
```

Here's an example with Intel iGPU
```
level=INFO source=common.go:49 msg="Dynamic LLM libraries" runners="[cpu_avx cpu_avx2 oneapi cpu]"
018+01:00 level=INFO source=gpu.go:221 msg="looking for compatible GPUs"
122+01:00 level=INFO source=types.go:123 msg="inference compute" id=0 library=oneapi variant="" compute="" driver=0.0 name="Intel(R) UHD Graphics" total="0 B" available="0 B"
```

**Reply by Deekerman** [reply]
Well it does seem to find it, i dont know if its now or before but thats good its there. 

```
Oct 29 15:31:34 openwebui ollama[2742]: time=2024-10-29T15:31:34.431-04:00 level=INFO source=common.go:49 msg="Dynamic LLM libraries" runners="[cuda_v12 rocm_v60102 cpu cpu_avx cpu_avx2 cuda_v11]"
Oct 29 15:31:34 openwebui ollama[2742]: time=2024-10-29T15:31:34.432-04:00 level=INFO source=gpu.go:221 msg="looking for compatible GPUs"
Oct 29 15:31:34 openwebui ollama[2742]: time=2024-10-29T15:31:34.637-04:00 level=INFO source=types.go:123 msg="inference compute" id=GPU-removed library=cuda variant=v12 compute=6..1 driver=12.4 name="Quadro P2000" total="4.9 GiB" available="4.7 GiB"

```

When i run a test, i guess i was expecting to see something pop up here, indicating it was used. 
Im still pretty new to this so maybe im understanding it wrong. 

<img width="410" alt="image" src="https://github.com/user-attachments/assets/a2f39179-cb09-4f01-b9c1-a792463f3f9e">


Thank you for your help.

**Reply by mhoogenbosch** [reply]
> Well it does seem to find it, i dont know if its now or before but thats good its there.
> 
> ```
> Oct 29 15:31:34 openwebui ollama[2742]: time=2024-10-29T15:31:34.431-04:00 level=INFO source=common.go:49 msg="Dynamic LLM libraries" runners="[cuda_v12 rocm_v60102 cpu cpu_avx cpu_avx2 cuda_v11]"
> Oct 29 15:31:34 openwebui ollama[2742]: time=2024-10-29T15:31:34.432-04:00 level=INFO source=gpu.go:221 msg="looking for compatible GPUs"
> Oct 29 15:31:34 openwebui ollama[2742]: time=2024-10-29T15:31:34.637-04:00 level=INFO source=types.go:123 msg="inference compute" id=GPU-removed library=cuda variant=v12 compute=6..1 driver=12.4 name="Quadro P2000" total="4.9 GiB" available="4.7 GiB"
> ```
> 
> When i run a test, i guess i was expecting to see something pop up here, indicating it was used. Im still pretty new to this so maybe im understanding it wrong.
> 
> <img alt="image" width="410" src="https://private-user-images.githubusercontent.com/33733383/381281503-a2f39179-cb09-4f01-b9c1-a792463f3f9e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyMzIzMTIsIm5iZiI6MTczMDIzMjAxMiwicGF0aCI6Ii8zMzczMzM4My8zODEyODE1MDMtYTJmMzkxNzktY2IwOS00ZjAxLWI5YzEtYTc5MjQ2M2YzZjllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI5VDIwMDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5ZjJkNzBjNDM5NzE4NDBjNGE3MmU5MTRiYjcyOWNmNDVmM2UxMDE3ZmM3ZDc0MWI1Mzc3M2E4MDY1NDA3ZGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.u12i1AdYmqoiObbLsD96dN3uL99VBL9wpAhP2z6cLqg">
> Thank you for your help.

You need to run NVIDIA-SMI on the host. So in the proxmox ssh console, not on the LXC host. So maybe you're in luck and it just does work. Mine does, when checking the host, it shows the correct information. You need to 'ask' a question first.

![image](https://github.com/user-attachments/assets/db8735a4-c40f-4dc3-9059-c29dc9e591f6)

**Reply by Deekerman** [reply]
@mhoogenbosch well it does show up there!

Thank you for the clarification and assistance. I thought it would show up in the lxc console but again, new to this. I got my emby working and tdarr and basically left it alone since then. 

<img width="409" alt="image" src="https://github.com/user-attachments/assets/bca9207d-a80d-4a6e-b39b-4d556d8866f6">

**Comment by tristan-k** [comment]
Please see my comment at the [corresponding ollama issue](https://github.com/ollama/ollama/issues/1495#issuecomment-2445281499).

**Comment by Popcorncandy09** [comment]
how can i setup SSO with authentik?
